website/blog/2023-04-20-LLM-tuning-math/index.mdx

```diff
@@ -0,0 +1,69 @@
+
+We will use the new subpackage [`flaml.autogen`](docs/Use-Cases/Auto-Generation) to automatically find the best model and inference parameter for LLMs on a given task and dataset given an inference budget, using a novel low-cost search & pruning strategy. FLAML currently supports all the LLMs from OpenAI, such as GPT-3.5 and GPT-4.
+
+We will use FLAML to perform model selection and inference parameter tuning. Then we compare the performance and inference cost on solving algebra problems with the untuned gpt-4. We will also analyze how different difficulty levels affect the results.
+
+## Experiment Setup
+
+We use FLAML to select between the following models with a target inference budget $0.02 per instance:
+- gpt-3.5-turbo, a relatively cheap model that powers the popular ChatGPT app
+- gpt-4, the state of the art LLM that costs more than 100 times of gpt-3.5-turbo
+
+We adapt the models using 20 examples in the train set, using the problem statement as the input and generating the solution as the output. We use the following inference parameters:
+
+- temperature: The parameter that controls the randomness of the output text. A higher temperature means more diversity but less coherence. We search for the optimal temperature in the range of [0, 1].
+- top_p: The parameter that controls the probability mass of the output tokens. Only tokens with a cumulative probability less than or equal to top-p are considered. A lower top-p means more diversity but less coherence. We search for the optimal top-p in the range of [0, 1].
+- max_tokens: The maximum number of tokens that can be generated for each output. We search for the optimal max length in the range of [50, 1000].
+- n: The number of responses to generate. Generating multiple responses and select the best among them can help with resolving potential errors due to randomness. In this experiment, we find the answer with highest votes among all the responses and then select it as the final answer to compare with the ground truth. For example, if n=5 and 3 of the responses contain a final answer 301 while 2 of the responses contain a final answer 159, we choose 301 as the final answer.
+We search for the optimal n in the range of [1, 100].
+- prompt: We use the template: "{problem} Solve the problem carefully. Simplify your answer as much as possible. Put the final answer in \\boxed{{}}." where {problem} will be replaced by the math problem instance.
+
+We use the average accuracy and inference cost as the metric to evaluate the performance. The inference cost is measured by the price per 1K tokens and the number of tokens consumed.
```

Comment:
```suggestion
In this experiment, when n > 1, we find the answer with the highest votes among all the responses and then select it as the final answer to compare with the ground truth. For example, if n=5 and 3 of the responses contain a final answer 301 while 2 of the responses contain a final answer 159, we choose 301 as the final answer. This can help with resolving potential errors due to randomness. We use the average accuracy and inference cost as the metric to evaluate the performance. The inference cost is measured by the price per 1K tokens and the number of tokens consumed.
```